docs/the-git-based-process-you-need-for-mainframe-development-DRAFT/the-git-based-development-process-for-mainframe-development.md

```diff
@@ -0,0 +1,303 @@
+
+The usage of *epic* branches can compose various styles of development processes. The section 4.1 Development Process variations provides additional samples.
+
+## Workflows in this development strategy
+
+The model facilitates three different types of development workflows:
+
+-   With a single long-living branch, the development process allows developers to work and focus on the next planned release. After planning the work items for the next release, the development team is adding changes to the *main* branch. This is the default development workflow.
+
+This Git branching strategy also facilitates:
+
+-   resolution of a production problem in the currently-released version of the application by leveraging a *release* branch that is used for maintenance purposes,
+
+-   concurrent development activities for significant solution development initiatives, which include multiple planned work items for a later delivery (including starting development of a future release) by creating an *epic* branch from a commit point in the history of *main*.
+
+Git tags are used throughout this process to indicate and label important commits, like the commit of a release that is built from the *main* branch, or a maintenance release created from a *release* maintenance branch.
+
+The next sections outline the various tasks and activities performed by the development team in the context of the above three scenarios.
+
+### Deliver changes with the next planned release 
+
+The below diagram depicts the typical workflow to deliver changes for the next planned release. In the default workflow, the development team commits changes to the head of the main branch. The changes of the next planned release are built, packaged, and released off the main branch.
+
+![Figure 3 - Git branching workflow supporting a Release-based delivery approach](images/media/image4.png)
+
+Developers implement their changes by committing to short-living feature branches (visualized in yellow) and integrate those via Pull Requests into the long-living *main* branch, which is configured to be a protected branch.
+
+At a high level, the development team works through the following tasks:
+
+1.  New work items are managed in the backlog. The team decides which work items will be implemented in the next iteration. Each application team can decide about the duration of the iteration (which can also be seen as the development cycle). In the above diagram, three work items were selected to be implemented for the next iteration. The development team is responsible for coordinating if features are demanding to be implemented in a specific order.
+
+2.  For each work item, a feature branch is created according to pre-defined naming conventions, allowing the assigned developers to have a copy of the codebase on which they can work in isolation from other concurrent development activities.
```

Comment:
I always feel 'copy' is a bit of an ambiguous word in the git context. 

I know this sentence means a copy in the sense of a branch in the repo, but the clones made by developers when they work in their IDEs are also 'copies'.

Would it work equally well without using the phrase including 'copy'?

> ... allowing the assigned developers to work in isolation from other development activities.

Reply:
... and actually, as witness of my unease, the very next point says:

> ... developers create a copy of the Git repository on their local workstations through the clone operation of Git

Reply:
First pass here - https://github.com/IBM/dbb-zappbuild/pull/420